lib/src/main/java/uk/gov/di/data/lep/library/dto/DeathNotification/DeathNotificationSetMapper.java

```diff
@@ -0,0 +1,138 @@
+import java.util.stream.Stream;
+
+public class DeathNotificationSetMapper {
+    public static DeathNotificationSet generateDeathNotificationSet(GroJsonRecord groRecord) {
+        var newEvent = generateDeathRegistrationEvent(groRecord);
+
+        var isUpdate = groRecord.recordLockedDateTime() == null;
+        var events = new DeathRegistrationEventMapping(
+            isUpdate ? null : newEvent,
+            isUpdate ? newEvent : null
+        );
+        var iat = Instant.now().getEpochSecond();
+        var jti = UUID.randomUUID().toString();
+        var toe = isUpdate
+            ? groRecord.recordUpdateDateTime().toEpochSecond(ZoneOffset.UTC)
+            : groRecord.recordLockedDateTime().toEpochSecond(ZoneOffset.UTC);
+        var txn = UUID.randomUUID().toString();
+
+        return new DeathNotificationSet(
+            null,
+            events,
+            null,
+            iat,
+            null,
+            jti,
+            null,
+            null,
+            toe,
+            txn
+        );
+    }
+
+    private static DeathRegistrationEvent generateDeathRegistrationEvent(GroJsonRecord groJsonRecord) {
+        var dateOfDeath = groJsonRecord.deceasedDeathDate().personDeathDate() == null
+            ? LocalDate.of(groJsonRecord.partialYearOfDeath(), groJsonRecord.partialMonthOfDeath(), 1)
+            : groJsonRecord.deceasedDeathDate().personDeathDate();
+
+        var deathDate = new IsoDate(groJsonRecord.qualifierText(), dateOfDeath);
+        var deathRegistrationID = groJsonRecord.registrationId();
+        var deathRegistrationUpdateReason = DeathRegistrationUpdateReasonType.fromGroRegistrationType(groJsonRecord.registrationType());
+        var freeFormatDeathDate = groJsonRecord.freeFormatDeathDate();
+        var recordUpdateTime = new StructuredDateTime(groJsonRecord.recordUpdateDateTime());
+        var deathRegistrationTime = new StructuredDateTime(groJsonRecord.recordLockedDateTime());
+
+        return new DeathRegistrationEvent(
+            deathDate,
+            deathRegistrationID,
+            deathRegistrationUpdateReason,
+            freeFormatDeathDate,
+            recordUpdateTime,
+            deathRegistrationTime,
+            generateDeathRegistrationSubject(groJsonRecord)
+        );
+    }
+
+    private static DeathRegistrationSubject generateDeathRegistrationSubject(GroJsonRecord groJsonRecord) {
+        var address = new PostalAddress(
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            groJsonRecord.deceasedAddress().postcode(),
+            null,
+            null,
+            null,
+            null,
+            null
+        );
+        var dateOfBirth = groJsonRecord.deceasedBirthDate().personBirthDate() == null
+            ? LocalDate.of(groJsonRecord.partialYearOfBirth(), groJsonRecord.partialMonthOfBirth(), 1)
```

Comment:
As discussed before, we probably actually want this to be a TemporalAccessor. It's conceivable that "partialMonthOfBirth" is null too, and we should handle that case